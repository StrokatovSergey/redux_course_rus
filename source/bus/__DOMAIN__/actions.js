import {types} from './types';

export const actions = {
	action: () => ({
		type : types.TYPE
	})
}