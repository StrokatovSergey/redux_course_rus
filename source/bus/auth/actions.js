import {types} from './types';

export const authAction = {
	authenticate: () => ({
		type : types.AUTHENTICATE
	}),
	signupAsync: (userData) => ({
		types: types.SIGNUP_ASYNC,
		payload: userData
	})
}