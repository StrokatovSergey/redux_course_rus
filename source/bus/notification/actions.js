import {types} from './types';
import {v4} from 'uuid';

export const notificationActions = {
	showNotification: (message, type = 'info', source) => ({
		type : types.SHOW_NOTIFICATION,
		payload: {message, type, source, id: v4() }
	}),
	hideNotification: () => ({
		type : types.HIDE_NOTIFICATION
	})
}