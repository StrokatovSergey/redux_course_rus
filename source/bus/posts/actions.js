import {types} from './types';
import {api} from '../../REST';

export const postsActions = {
	fillPosts : (posts) => ({
		type: types.FILL_POSTS,
		payload: posts
	}),
	createPost : (comment) => ({
		type: types.CREATE_POST,
		payload: comment
	}),
	createPostAsync : (post) => ({
		type: types.CREATE_POST_ASYNC,
		payload: post
	}),
	fetchPostsAsync : () => async (dispatch, getState) => {
		dispatch({
			type: types.FETCH_POSTS_ASYNC
		})
		const responce = await api.posts.fetch()
		const result = await responce.json()

		dispatch(postsActions.fillPosts(result.data))
	}
}