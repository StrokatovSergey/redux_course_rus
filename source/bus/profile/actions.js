import {types} from './types';

export const profileActions = {
	fillProfile: (profile) => ({
		type : types.FILL_PROFILE,
		payload: profile
	}),
	clearProfile: () => ({
		type : types.CLEAR_PROFILE,
	}),
	updateAvatar: (newAvatarUrl) => ({
		type : types.UPDATE_AVATAR,
		payload: newAvatarUrl
	}),


	updateNameAsync: (newName) => ({
		type : types.UPDATE_NAME_ASYNC,
		payload: newName
	}),
	updateAvatarAsync: (newAvatarUrl) => ({
		type : types.UPDATE_AVATAR_ASYNC,
		payload: newAvatarUrl
	}),
	updatePasswordAsync: (passwordData) => ({
		type : types.UPDATE_PASSWORD_ASYNC,
		payload: passwordData
	}),
}