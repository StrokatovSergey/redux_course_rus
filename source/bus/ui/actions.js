import {types} from './types';

export const uiActions = {
	startFetching: () => ({
		type : types.START_FETCHING
	}),
	stopFetching: () => ({
		type : types.STOP_FETCHING
	}),
	emitError: (error, meta = null) => ({
		type: types.EMIT_ERROR,
		payload: error,
		error: true,
		meta
	})
}