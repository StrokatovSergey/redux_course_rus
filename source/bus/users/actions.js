import {types} from './types';

export const usersActions = {
	fillUsers: (users) => ({
		type : types.FILL_USERS,
		payload: users
	}),
	clearUsers: () => ({
		type : types.CLEAR_USERS,
	}),
	fetchUsersAsync: () => ({
		type : types.FETCH_USERS_ASYNC,
	})
}