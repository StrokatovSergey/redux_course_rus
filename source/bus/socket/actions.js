import { socket } from '../../init/socket';
import { uiActions } from '../ui/actions';
import {postsActions} from '../posts/actions';
import {store} from '../../init/store';

export const socketActions = {
	listenConnection: () => dispatch => {
		socket.on('connect', () => {
			dispatch(uiActions.setOnlineState())
		});

		socket.on('disconnect', () => {
			dispatch(uiActions.setOfflineState())
		});
	},
	listenPosts: () => (dispatch, getState) => {
		socket.on('create', event => {
			const {data: post} = JSON.parse(event)
			dispatch(postsActions.createPost(post))
		})

		socket.on('like', event => {
			const {data, meta} = JSON.parse(event)
			const liker = getState().users.find(user => user.get('id') === data.userId).delete('avatar')
			if (meta.action === 'like') {
			   dispatch( postsActions.likePost({postId: data.postId, liker}))
			} else {
				dispatch(postsActions.unlikePost({postId: data.postId, liker}))
			}
		})

		socket.on('remove', event => {
			const { data: postId } = JSON.parse(event)
			dispatch(postsActions.removePost(postId))
		})
	}
}