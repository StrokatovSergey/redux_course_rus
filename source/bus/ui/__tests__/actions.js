import { uiActions } from "../actions";

describe("actions profile", () => {
  it("should startFetching snapshot", function() {
    expect(uiActions.startFetching()).toMatchInlineSnapshot(`
Object {
  "type": "START_FETCHING",
}
`);
  });

  it("should stopFetching snapshot", function() {
    expect(uiActions.stopFetching()).toMatchInlineSnapshot(`
Object {
  "type": "STOP_FETCHING",
}
`);
  });

  it("should setOnlineState snapshot", function() {
    expect(uiActions.setOnlineState()).toMatchInlineSnapshot(`
Object {
  "type": "SET_ONLINE_STATE",
}
`);
  });

  it("should setOfflineState snapshot", function() {
    expect(uiActions.setOfflineState()).toMatchInlineSnapshot(`
Object {
  "type": "SET_OFFLINE_STATE",
}
`);
  });

  it("should setOfflineState snapshot", function() {
    expect(uiActions.emitError()).toMatchInlineSnapshot(`
Object {
  "error": true,
  "meta": null,
  "payload": undefined,
  "type": "EMIT_ERROR",
}
`);
  });
});
