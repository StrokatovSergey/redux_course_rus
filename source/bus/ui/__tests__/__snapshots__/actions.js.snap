// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions profile should clearProfile snapshot 1`] = `
Object {
  "type": "CLEAR_PROFILE",
}
`;

exports[`actions profile should fillProfile snapshot 1`] = `
Object {
  "payload": Object {
    "avatar": "TEST_AVATAR",
    "firstName": "Walter",
    "id": "TEST_ID",
    "lastName": "White",
    "token": "TEST_TOKEN",
  },
  "type": "FILL_PROFILE",
}
`;

exports[`actions profile should udpateAvatar snapshot 1`] = `
Object {
  "payload": "https://www.url.com",
  "type": "UPDATE_AVATAR",
}
`;

exports[`actions profile should updateAvatarAsync snapshot 1`] = `
Object {
  "payload": Array [
    "avatar",
  ],
  "type": "UPDATE_AVATAR_ASYNC",
}
`;

exports[`actions profile should updateNameAsync snapshot 1`] = `
Object {
  "payload": Object {
    "firstName": "Mister",
    "lastName": "Twister",
  },
  "type": "UPDATE_NAME_ASYNC",
}
`;

exports[`actions profile should updatePasswordAsync snapshot 1`] = `
Object {
  "payload": Object {
    "newPassword": 1234567,
    "oldPassword": 12345,
  },
  "type": "UPDATE_PASSWORD_ASYNC",
}
`;
